txgh.gemspec

```diff
@@ -0,0 +1,27 @@
+$:.unshift File.join(File.dirname(__FILE__), 'lib')
+require 'txgh/version'
+
+Gem::Specification.new do |s|
+  s.name     = 'txgh'
+  s.version  = ::Txgh::VERSION
+  s.authors  = ['Matthew Jackowski', 'Cameron Dutro']
+  s.email    = ['mattjjacko@gmail.com', 'camertron@gmail.com']
+  s.homepage = "https://github.com/transifex/txgh"
+
+  s.description = s.summary = 'A server that automatically syncs translation resources between Github and Transifex.'
+
+  s.platform = Gem::Platform::RUBY
+  s.has_rdoc = true
+
+  s.add_dependency 'faraday', '~> 0.9'
+  s.add_dependency 'faraday_middleware', '~> 0.10'
+  s.add_dependency 'json', '~> 1.8'
+  s.add_dependency 'octokit', '~> 4.2'
+  s.add_dependency 'puma', '~> 2.15'
+  s.add_dependency 'parseconfig', '~> 1.0'
+  s.add_dependency 'sinatra', '~> 1.4'
+  s.add_dependency 'sinatra-contrib', '~> 1.4'
```

Comment:
I definitely agree with keeping your dependencies up to date! No doubt it can be a big source of code debt.

I'm not sure it's a good idea to remove the version constraints from these gem dependencies however. As I'm sure you know, rubygems are supposed to follow [SemVer](http://semver.org/), or semantic versioning (maybe python packages do this as well?) Version numbers have three parts: major.minor.patch. The `~>` in gem/bundler parlance means "allow the last specified number to go up". In other words, if I specify the gem `faraday '~> 0.9'`, bundler will install the highest _minor_ version of faraday available. It might install faraday 0.10, or faraday 0.9.1, or even faraday 0.10.1. It will _not_ install faraday 1.0 because that would constitute a _major_ version change, which are breaking, backwards-incompatible changes. By locking gem dependencies to minor versions, you avoid breaking your code accidentally.

I know what you're thinking: nightly builds should catch any issues major version bumps might cause. While that's true for txgh as a stand-alone project, it might have adverse affects for users. Imagine the scenario where a developer includes txgh in their Gemfile (or gemspec) along with a number of other dependencies. Let's say they include faraday as a dependency, so now their gemfile looks like this:

```ruby
gem 'faraday', '~> 0.9'
gem 'txgh', '~> 1.0'
```

A new version of faraday, v1.0, has just been released. Since txgh doesn't specify a version requirement for faraday, bundler will happily install the 0.9 version. Txgh was developed against the 1.0 version of faraday because the Travis nightly build notified us the faraday interface had changed. Unfortunately this now puts the developer in a tough spot: they either have to upgrade faraday or downgrade txgh. If txgh had specified a version of faraday, this issue could have been caught on `bundle install`.